sql/src/main/java/org/apache/druid/sql/calcite/planner/Calcites.java

```diff
@@ -397,4 +402,44 @@ public static int collapseFetch(int innerFetch, int outerFetch, int outerOffset)
     }
     return fetch;
   }
+
+  public static Class<?> sqlTypeNameJdbcToJavaClass(SqlTypeName typeName)
+  {
+    JDBCType jdbcType = JDBCType.valueOf(typeName.getJdbcOrdinal());
+    switch (jdbcType) {
+      case CHAR:
+      case VARCHAR:
+      case LONGVARCHAR:
+        return String.class;
+      case NUMERIC:
+      case DECIMAL:
+        return BigDecimal.class;
+      case BIT:
+        return Boolean.class;
+      case TINYINT:
+        return Byte.class;
+      case SMALLINT:
+        return Short.class;
+      case INTEGER:
+        return Integer.class;
+      case BIGINT:
+        return Long.class;
+      case REAL:
+        return Float.class;
+      case FLOAT:
```

Comment:
I think a comment for why this is not `Float.class` would be helpful

Reply:
Added link to docs I used as reference for mappings from JDBC to Java types